federated-server/shared-state.js

```diff
@@ -0,0 +1,50 @@
+
+const SharedState = require("shared-state-js");
+const { strMapToObj, objToStrMap } = require('obj2map');
+const fetch = require('node-fetch');
+const pull = require('pull-stream')
+const { genericId } = require('./config')
+const { flatToUnique } = require("./utils/faltToUnique");
+
+// Create a shared-state object
+const myState = new SharedState({author: 'soporte-remoto'})
+
+const requestSharedState = (table, sync) => {
+    const hostUrl = `http://thisnode.info/cgi-bin/shared-state/${table}`;
+    return fetch(hostUrl, {
+        method: 'POST',
+        body: sync? JSON.stringify(strMapToObj(myState.show(table))) : []
+    })
+    .then(res => res.json())
+    .catch(err => console.log('Http merge error', err));
+}
+
+
+export async function getActualNodes(){
+    //Merge node data into my shared-state
+    const nodeData = await requestSharedState('babeld-hosts', false);
+    myState.merge(objToStrMap(nodeData) ,false,'babeld-hosts');
+    //Format result to get a hostname list
+    let nodes = []
+    myState
+        .show('babeld-hosts')
+        .forEach(babledHost => nodes = nodes.indexOf(babledHost.data) !== -1 ? nodes: [...nodes, babledHost.data])
```

Comment:
There is a typo here: babledHost. And also I just checked that we have [Sets](http://javascript.info/map-set#set) in JS, if you want to use it as a shortcut.

Reply:
done